web/src/components/navbar/desktop/NavBarDesktop.tsx

```diff
@@ -0,0 +1,185 @@
+import { ButtonIcon } from "@/components/ButtonIcon";
+import { NavBarLogoOnly } from "@/components/navbar/NavBarLogoOnly";
+import { DesktopLoginButton } from "@/components/navbar/desktop/DesktopLoginButton";
+import { NavBarDesktopDropDown } from "@/components/navbar/desktop/NavBarDesktopDropDown";
+import { NavBarDesktopHomeLogo } from "@/components/navbar/desktop/NavBarDesktopHomeLogo";
+import { NavBarDesktopQuickLinks } from "@/components/navbar/desktop/NavBarDesktopQuickLinks";
+import { NavBarVerticalLineDivider } from "@/components/navbar/desktop/NavBarDesktopVerticalLineDivider";
+import { NavBarDesktopWrapper } from "@/components/navbar/desktop/NavBarDesktopWrapper";
+import {
+  AddBusinessItem,
+  GetStartedMenuItem,
+  LogoutMenuItem,
+  MyNjMenuItem,
+  ProfileMenuItem,
+  RegisterMenuItem,
+} from "@/components/navbar/shared-submenu-components";
+import { Icon } from "@/components/njwds/Icon";
+import { useConfig } from "@/lib/data-hooks/useConfig";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { getBusinessIconColor } from "@/lib/domain-logic/getBusinessIconColor";
+import { getNavBarBusinessTitle } from "@/lib/domain-logic/getNavBarBusinessTitle";
+import { orderBusinessIdsByDateCreated } from "@/lib/domain-logic/orderBusinessIdsByDateCreated";
+import React, { ReactElement, useEffect, useRef, useState } from "react";
+
+interface Props {
+  previousBusinessId?: string | undefined;
+  isLanding: boolean | undefined;
+  logoOnlyType?: "NAVIGATOR_LOGO" | "NAVIGATOR_MYNJ_LOGO" | undefined;
+  currentlyOnboarding: boolean | undefined;
+  isAuthenticated: boolean;
+}
+
+export const NavBarDesktop = (props: Props): ReactElement => {
+  const { business, userData } = useUserData();
+  const { Config } = useConfig();
+
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef<HTMLButtonElement | null>(null);
+
+  const handleClose = (
+    event?: MouseEvent | TouchEvent | React.MouseEvent<HTMLLIElement> | React.MouseEvent<Document>
+  ): void => {
+    if (event && anchorRef.current && anchorRef.current.contains(event.target as Node)) {
+      return;
+    }
+
+    setOpen(false);
+  };
+
+  const prevOpen = useRef(open);
+  useEffect(() => {
+    if (null !== anchorRef.current && prevOpen.current && !open) {
+      anchorRef.current.focus();
+    }
+    prevOpen.current = open;
+  }, [open]);
+
+  const textColor = props.isAuthenticated ? "primary" : "base";
+  const navBarBusinessTitle = getNavBarBusinessTitle(business, props.isAuthenticated);
+  const currentIndex =
+    userData && business ? orderBusinessIdsByDateCreated(userData).indexOf(business.id) : 0;
+
+  if (props.logoOnlyType) {
+    // loading/redirect/ethan onboarding
+    return <NavBarLogoOnly logoType={props.logoOnlyType} />;
```

Comment:
sand - So this component could and probably should be refactored into the same pattern as below (I will write a ticket to do so)